extends Node2D

const SPEED = 60

var direction = 1
@onready var animated_sprite = $AnimatedSprite2D
@onready var ray_cast_right = $RayCastRight
@onready var ray_cast_left = $RayCastLeft

# Called when the node enters the scene tree for the first time.
func _ready():
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta): #delta = time elapsed between frames. Pretty much multiply it by differenc in location to account for FPS
	if ray_cast_right.is_colliding():
		direction = -1
		animated_sprite.flip_h = true
	if ray_cast_left.is_colliding():
		direction = 1
		animated_sprite.flip_h = false
	position.x += SPEED * direction * delta #speed is speed, direction's absolute value = 1, negative or postive to determine value of speed
